src/AdminBarApi.php

```diff
@@ -7,36 +7,96 @@
  */
 class AdminBarApi extends AbstractEndpoint {
 
-	/**
-	 * Endpoint path
-	 *
-	 * @Override
-	 * @var String
-	 */
-	protected $endpoint = '/admin-bar';
-
-	/**
-	 * Get the data.
-	 *
-	 * @Override
-	 * @param \WP_REST_Request $request The request.
-	 *
-	 * @return array|\WP_Error
-	 */
-	public function endpoint_callback( \WP_REST_Request $request ) {
-		global $wp_admin_bar;
-
-		$current_user = wp_get_current_user();
-
-		$data = [
-			'site_name' => get_bloginfo( 'name' ),
-			'user_name' => $current_user->user_login,
-			'dashboard_url' => admin_url(),
-			'logout_url' => wp_logout_url(),
-			'edit_page_url' => '',
-			'widgets' => $wp_admin_bar->get_node( 'new-content' ),
-		];
-
-		return $this->filter_data( $data );
-	}
+  /**
+   * Endpoint path
+   *
+   * @Override
+   * @var String
+   */
+  protected $endpoint = '/admin-bar';
+
+  /**
+   * Cookie name
+   *
+   * @var String
+   */
+  protected $cookieName = 'wp_admin_bar_user'; 
```

Comment:
@mitogh 

As I understand each user login event will trigger the login_action, and that action will generate the cookie on the client side with the user name as value. So every user that logs in to WP will have the wp_admin_bar_user cookie stored in their browsers, but each of them will have their own value.